src/main/java/org/hobbit/core/components/AbstractCommandReceivingComponent.java

```diff
@@ -279,18 +280,37 @@ protected String createContainer(String imageName, String[] envVariables) {
      */
     protected String createContainer(String imageName, String containerType, String[] envVariables) {
         try {
+            return createContainerAsync(imageName, containerType, envVariables).get();
+        } catch (Exception e) {
+            LOGGER.error("Got exception while trying to request the creation of an instance of the \"" + imageName
+                    + "\" image.", e);
+        }
+        return null;
+    }
+
+    protected Future<String> createContainerAsync(String imageName, String containerType, String[] envVariables) {
+        try {
+            SettableFuture<String> containerFuture = SettableFuture.create();
+
             envVariables = extendContainerEnvVariables(envVariables);
 
-            initResponseQueue();
+            String responseQueueName = cmdChannel.queueDeclare().getQueue();
```

Comment:
Creating a new queue for each request could make the container creation more expensive :disappointed: